src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -720,7 +720,7 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType, bool inExpressionTr
 
             if (returnType.HasType)
             {
-                if (returnType.Type.ContainsNativeInteger())
+                if (compilation.ShouldEmitNativeIntegerAttributes() && returnType.Type.ContainsNativeInteger())
```

Comment:
>compilation.ShouldEmitNativeIntegerAttributes() && returnType.Type.ContainsNativeInteger() [](http://example.com/codeflow?start=20&length=90)

It looks like this pattern is repeated a number of times. Consider adding a `static bool ShouldEmitNativeIntegerAttributes(this CSharpCompilation, TypeSymbol type)` overload combines the two.

Reply:
I like that :-) I'm tempted to do it for `NeedsNullableAttribute` below too, but I'll leave that alone...